cli/tests/unit/net_test.ts

```diff
@@ -325,6 +327,55 @@ unitTest(
   },
 );
 
+unitTest(
+  { permissions: { net: true } },
+  async function netUdpSendReceiveBroadcast() {
+    const alice = Deno.listenDatagram({ port: 3500, transport: "udp" });
+    assert(!alice.broadcast);
+
+    const bob = Deno.listenDatagram({
+      port: 4501,
+      transport: "udp",
+      hostname: "0.0.0.0",
+    });
+    assert(bob.addr.transport === "udp");
+    assertEquals(bob.addr.port, 4501);
+    assertEquals(bob.addr.hostname, "0.0.0.0");
+
+    const broadcastAddr = { ...bob.addr, hostname: "255.255.255.255" };
+
+    const sent = new Uint8Array([1, 2, 3]);
+
+    let acceptErrCount = 0;
+    const checkErr = (e: Error) => {
+      if (e instanceof Deno.errors.PermissionDenied) {
+        // Broadcasting without SO_BROADCAST set should result in a permissions error.
+        acceptErrCount++;
+      } else {
+        throw new Error("Unexpected error message: " + e.message);
+      }
+    };
+    // Sending to the broadcast addr without broadcast flag should result in error.
+    await alice.send(sent, broadcastAddr).catch(checkErr);
+
+    alice.setBroadcast(true);
+    assert(alice.broadcast);
+    const byteLength = await alice.send(sent, broadcastAddr);
```

Comment:
This is the line that's failing on MacOS:
```
AddrNotAvailable: Can't assign requested address (os error 49)
```
Trying to figure out why this fails--I'm guessing it may be related to the IP addresses bound in this code. But I don't have access to a Mac currently to try and debug.

(The subsequent test fails because we do not close the socket)

Reply:
Interestingly broadcasting from 0.0.0.0 to 255.255.255.255 WFM locally but I'm still on macos 10.14 so maybe that has something to do with it.

I'm okay with skipping the test on macos but please add a comment explaining why.

Reply:
This is actually resolved now; the broken version of this unit test was not specifying a hostname at all when binding the sender--so it was using the default hostname option. `0.0.0.0` indeed seems to fix it. Awkwardly, this unit test will probably send a real UDP broadcast packet out on the network of whoever runs it though.